src/Database/Connection.php

```diff
@@ -727,6 +728,39 @@ protected function _newLogger(StatementInterface $statement)
     }
 
     /**
+     * Checks if the driver supports EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @return bool
+     */
+    public function canExplain($query)
+    {
+        $query = (string)$query;
+
+        return preg_match('/^\s*(SELECT|INSERT|UPDATE|DELETE)\s/i', $query)
+            && $this->_driver->explainSQL($query) !== false;
+    }
+
+    /**
+     * Executes an EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @param array $params list or associative array of params to be interpolated in $query as values
+     * @param array $types list or associative array of types to be used for casting values in query
+     * @return \Cake\Database\StatementInterface
+     */
+    public function explain($query, array $params = [], array $types = [])
```

Comment:
@markstory OK. Then I would like to discard all commits on this PR, except for changes of `LoggedQuery`, as I don't think such a `explainSQL()` is useful if it returns multiple queries. Instead, I would simply prepend a `EXPLAIN` keyword to the logged query in DebugKit, without abstraction. Even so, probably people could hook the `ToolbarController::sqlExplain()` action during a `Controller.startup` event, if they want to support Oracle or other drivers.